app/controllers/users/emails_controller.rb

```diff
@@ -17,9 +17,10 @@ def show
     end
 
     def add
+      request_id = sp_session[:request_id] || nil
```

Comment:
`|| nil` probably doesn't do anything here if we're already expecting the value to be `nil`?

```suggestion
      request_id = sp_session[:request_id]
```